sys/include/timex.h

```diff
@@ -31,32 +31,32 @@ extern "C" {
 /**
  * @brief The number of microseconds per second
  */
-#define US_PER_SEC (1000000U)
+#define US_PER_SEC ((uint32_t)1000000U)
```

Comment:
Can we please use `(1000000LU)`?

Reply:
I tried that, but that yielded 100+ compilation errors as `PRIu32` expects `unsigned int`, not `unsinged long`. Even though they are the same. But the cast clearly doesn't work, as other code is using `US_PER_SEC` in preprocessor `#if` statements. They don't work with types...

Reply:
Can you keep the `UL` here but add the cast where needed in `tests/xtimer_msg` ?

Reply:
Let me first try to weasel around this by using `U` on systems with `sizeof(unsinged) == `sizeof(uint32_t)` and `LU` on other systems. With using `LU` unconditionally, the number of compilation issues was huge...

Also: I think users can rightfully expect that `PRIu32` should work with something like e.g. `US_PER_SEC * 200`. The only way to solve this (except for the casts to `uint32_t`) is to make the postfix dependent on the integer width.

Reply:
OK. The current approach at least works locally. Let's wait for Murdock to confirm.

Reply:
uff, but it is mega hacky and different to everywhere else in the code.

Reply:
... but also quite contained and practical...

Reply:
Still fails (which I currently fail to understand). I need to think about this for some time. Maybe there is a cleaner solution that just didn't occur to me yet.

Reply:
OK. On ESP32 without an explicit `#include <limits.h>` is needed. (I was pretty sure that an `#include <stdint.h>` should have also provided `UINT_MAX`, but maybe I was wrong.)